spec/models/user_spec.rb

```diff
@@ -0,0 +1,72 @@
+require "rails_helper"
+require "./app/models/user"
+
+describe User do
+
+  context "when all attributes are correct" do
+
+    it "is created" do
+      user = User.create(name: "John Doe", email: "user@gmail.com", password: "secret", uid: "123")
+
+      expect(user).to be_a User
+    end
+  end
+
+  context "when password not given" do
+
+    it "raises an error" do
+      user = User.create(name: "John Doe", email: "user@gmail.com", uid: "123")
+
+      expect { user.save! }.to raise_error("Validation failed: Password can't be blank")
+    end
+  end
+
+  context "when name not given" do
+
+    it "raises an error" do
+      user = User.create(email: "user@gmail.com", password: "secret", uid: "123")
```

Comment:
You don't need to create an object in the database, which is expensive, in order to check for validation and error messages.

For example:

```rb
expect { User.new }.not_to be_valid
```

or

```rb
user = build(:user, name: nil)

user.valid?

expect(user.errors.full_messages).to include(...)
```

Ref: https://apidock.com/rails/ActiveModel/Errors/full_messages

```